### PR TITLE
Get started on the overfitting tip

### PR DESCRIPTION
**Pull request by Benjamin-Lee**
**Did you add yourself as a [contributor](https://github.com/Benjamin-Lee/deep-rules/blob/master/contributors.md) if this is your first contribution?**

- [x] Yes, I added myself or am already a contributor

**Any more details?**

### CODE REVIEW

**Comment by Benjamin-Lee** [comment]
> I think the student analogy should be dropped. Anthropomorphism and the use of "smarter" may suggest to some that DL is always the best option as long as you take care to not over fit (even though a previous tip addresses this explicitly)

While I agree that anthropomorphism is an issue, I don't think that precludes the use of analogies altogether. What about one or more of:

1. Clarifying the limitations of the anlogy
2. Switching to a non-human analogy
3. Using a visual such as this

**Comment by chevrm** [comment]
I like the idea of a visual representation of a well vs over fit model.

**Comment by rasbt** [comment]
I like the student analogy. 

There's also a general issue, which is related to overfitting but currently not be overcome with the methods we use. I.e., deep learning models have recently got to the state where they outperform humans in terms of detecting various diseases/medical conditions from images. However, this assumes that the cameras and angles of the pictures are similar to the ones used in the training and test dataset. If you you use a different system for taking this images is where the model may fall apart. 

If overfitting is related to generalization performance, where generalization is defined as Expectation[Loss(Y, h(X)]) and h is the hypothesis and Y and X are random variables, we assume we can only test a subset of X, Y, namely the test set to estimate the generalization error. However, that doesn't mean that even if the estimate of the generalization performance is satisfactory and the degree of overfitting is small, that the model does well on all medical images of a given object class. Or in other words, we may want to highlight the limitations of estimates of generalization performance as well!?

File: content/09.overfitting.md [hunk 1]
```diff
@@ -1 +1,17 @@
 ## Tip 7: Address deep neural networks' increased tendency to overfit the dataset {#overfitting}
+
+Overfitting is one of the most significant dangers faced by a deep learning practitioner.
+Put simply, overfitting is when a model–deep or otherwise–memorizes the mapping from inputs to outputs rather than truly learning.
+This subtle distinction is made clearer by seeing what happens when a model is tested on data to which it was not exposed during training: just as a student who memorizes exam materials struggles to correctly answer questions for which they have not studied, a machine learning model that has overfit to its training data will perform poorly on unseen test data.
+Deep learning models are particularly susceptible to overfitting due to their greater representational capacity.
+To continue the student analogy, a smarter student has greater potential for memorization than average one and thus may be more inclined to memorize.
+
+The simplest way to combat overfitting is to detect it.
+This can be done by splitting the dataset into three parts: a training set, a tuning set (also commonly called a validation set), and a testing set.
```
[the remark this answers is not in the document]
Comment (Benjamin-Lee) [reply content/09.overfitting.md]: > "test set" sounds more natural

I disagree since its a parallel construction with "tuning" and "training" data

File: content/09.overfitting.md [hunk 2]
```diff
@@ -12,8 +12,8 @@ The simplest way to combat overfitting is to detect it.
...
 While a slight drop in performance from the training set to the test set is normal, a significant drop is a clear sign of overfitting.
-Additionally, there are a variety of techniques to reduce overfitting during training including data augmentation, regularization, dropout [@url:http://dl.acm.org/citation.cfm?id=2627435.2670313] and weight decay [@url:http://dl.acm.org/citation.cfm?id=2986916.2987033].
+Additionally, there are a variety of techniques to reduce overfitting during training including data augmentation and regularization techniques such as dropout [@url:http://dl.acm.org/citation.cfm?id=2627435.2670313] and weight decay [@url:http://dl.acm.org/citation.cfm?id=2986916.2987033].
 
-Additionally, one must be sure that their data are not skewed or biased, such as by having confounding variables that the model can pick up on [@arxiv:1807.00431](https://arxiv.org/abs/1807.00431)].
+Additionally, one must be sure that their data are not skewed or biased, such as by having confounding and scientifically non-relevant variables that the model can pick up on [@arxiv:1807.00431](https://arxiv.org/abs/1807.00431)].
```
Comment (rasbt) [inline content/09.overfitting.md]: Maybe we could phrase "skewed or biased, such as by having" & "the model can pick up on" a bit more elegantly. 

How about sth like this?

"Additionally, one must ensure that the datasets are not skewed, biased, or include confounding and scientifically non-relevant variables that the models can exploit to learn spurious input--output relationships."

File: content/09.overfitting.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (rasbt) [reply content/09.overfitting.md]: sure, but I think here these don't act as gerunds but rather technical terms. Never seen "training and independent testing sets" -- usually it's "training and independent test set". But this is a minor thing and either way is fine since both ways are valid for different reasons :P

File: content/09.overfitting.md (hunk 2 above)
Comment (agitter) [reply content/09.overfitting.md]: Citation update and I suggest "irrelevant" instead of "non-relevant":
```suggestion
Additionally, one must be sure that their data are not skewed or biased, such as by having confounding and scientifically irrelevant variables that the model can pick up on [@arxiv:1807.00431].
```

File: content/09.overfitting.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (Benjamin-Lee) [reply content/09.overfitting.md]: Well, it seems [Wikipedia is on your side](https://en.wikipedia.org/wiki/Training,_validation,_and_test_sets).

**Comment by Benjamin-Lee** [comment]
@rasbt I added a section mentioning the Chuang and Keiser baseline overfitting detection method

File: content/09.overfitting.md [hunk 3]
```diff
@@ -1 +1,21 @@
 ## Tip 7: Address deep neural networks' increased tendency to overfit the dataset {#overfitting}
+
+Overfitting is one of the most significant dangers faced by a deep learning practitioner.
+Put simply, overfitting occurs when a model fits patterns in the training data too closely, includes noise or non-scientifically relevant perturbations, or in the most extreme case, simply memorizes patterns in the training set.
+This subtle distinction is made clearer by seeing what happens when a model is tested on data to which it was not exposed during training: just as a student who memorizes exam materials struggles to correctly answer questions for which they have not studied, a machine learning model that has overfit to its training data will perform poorly on unseen test data.
+Deep learning models are particularly susceptible to overfitting due to their relatively large number of parameters and associated representational capacity.
+To continue the student analogy, a smarter student has greater potential for memorization than average one and thus may be more inclined to memorize.
```
Comment (agitter) [inline content/09.overfitting.md]: Like @chevrm, I think this analogy could be misleading.  If you and @rasbt like it, I'm fine leaving it in this draft, but I may want to revisit it later.

File: content/09.overfitting.md [hunk 4]
```diff
@@ -1 +1,21 @@
...
+
+![A visual example of overfitting. While a high-degree polynomial gets high accuracy on its training data, it performs poorly on data that is has not seen before, whereas a simple linear regression works well. The greater representational capacity of the polynomial is analogous to using a larger or deeper neural network.](images/overfitting.png){#fig:overfitting-fig}
```
Comment (agitter) [inline content/09.overfitting.md]: Can you please refer to this figure somewhere in the text using `Figure @fig:overfitting-fig`?

File: content/09.overfitting.md [hunk 5]
```diff
@@ -1 +1,21 @@
...
+
+The simplest way to combat overfitting is to detect it.
+This can be done by splitting the dataset into three parts: a training set, a tuning set (also commonly called a validation set), and a test set.
```
Comment (agitter) [inline content/09.overfitting.md]: How about `also commonly called a validation set in the machine learning literature` given the confusion about `validation` (#74).

File: content/09.overfitting.md (hunk 3 above)
Comment (Benjamin-Lee) [reply content/09.overfitting.md]: Please do! Let's get this merged and either you or @chevrm can PR changes/clarifications as y'all see fit.

File: content/09.overfitting.md [hunk 6]
```diff
@@ -1 +1,21 @@
...
+To continue the student analogy, a smarter student has greater potential for memorization than average one and thus may be more inclined to memorize.
+
+![A visual example of overfitting. While a high-degree polynomial gets high accuracy on its training data, it performs poorly on data that is has not seen before, whereas a simple linear regression works well. The greater representational capacity of the polynomial is analogous to using a larger or deeper neural network.](images/overfitting.png){#fig:overfitting-fig}
+
+The simplest way to combat overfitting is to detect it.
+This can be done by splitting the dataset into three parts: a training set, a tuning set (also commonly called a validation set in the machine learning literature), and a test set.
+By exposing the model solely to the training data during fitting, a researcher can use the model's performance on the unseen test data to measure the amount of overfitting.
+While a slight drop in performance from the training set to the test set is normal, a significant drop is a clear sign of overfitting (see Figure @fig:overfitting-fig for a visual demonstration of a overfit model that performs poorly on test data).
```
Comment (agitter) [inline content/09.overfitting.md]:
```suggestion
While a slight drop in performance from the training set to the test set is normal, a significant drop is a clear sign of overfitting (see Figure @fig:overfitting-fig for a visual demonstration of an overfit model that performs poorly on test data).
```
